### ISSUE TITLE
[vLLM plugin] Incorrect/random output generated with TP generative test

### ISSUE THREAD

**Issue by mmanzoorTT** [tenstorrent/tt-xla#4440]
Tensor parallel generative model tests are generating incorrect/random output for 2D mesh. This issue is caused by sampling on device. These tests generate correct results with either 1D mesh or 2D mesh with cpu sampling. 

Output for last few nightly runs for `Qwen/Qwen3-32B` with 2D mesh are below 
[CI job 27 Apr](https://github.com/tenstorrent/tt-xla/actions/runs/25026871133/job/73299881403#step:19:439)
```
prompt: I like taking walks in the, 
output:  restartgp erect workplaceSignpart Unblock_REGSamplerlekestado                                           (info Htmldom ///building neighborhoods somewhat '''iness(())
```
[CI job 26 Apr](https://github.com/tenstorrent/tt-xla/actions/runs/24970760261/job/73131748597#step:19:439)
```
prompt: I like taking walks in the, 
output:  dou调 arrangementyieldysisyieldSignatureited bitstern(\ THE Need(isset Handle Parameter arts post Gretaphael Handle Penguin surplus@SpringBootTestystem language
```
[CI job 25 Apr](https://github.com/tenstorrent/tt-xla/actions/runs/24944092006/job/73042387279#step:19:439)
```
prompt: I like taking walks in the, 
output:  restart twittererase workflow winnerfectiontreeimiento Regionpartizophren Son класс PureComponent '''class.PORT\Requests_LE volunte Bool Personen Concf typically Sony
```
[CI job 24 Apr](https://github.com/tenstorrent/tt-xla/actions/runs/24917761316/job/72973189022#step:19:439)
```
prompt: I like taking walks in the, 
output:  dou--){
hen(Main.PORTsession Unblock Unblock reslek artscoctract]=>SQL Unblock.cHaving_play availabilityestureRecognizer shares Unblock ''' Char_OPTION
```

**Comment by kmabeeTT** [comment]
## Bisect

First-bad: `346ef700c` (PR #3820, *[vLLM plugin] Add 2D mesh support*).

## Smaller repro

Qwen3-8B is the smallest model that reproduces (Qwen3-0.6B / 1.7B / 4B all generate coherent text with `use_2d_mesh=True` + device sampling). 32B takes ~20 min, 8B is far faster for iteration.

## Cause

`compute_logits` and `sample_from_logits` are separate `torch.compile`d graphs. The sampler reduces over the vocab dim (`sort` / `softmax` / `argmax(dim=-1)`) and assumes its input is fully replicated.

The 1D-mesh path enforces this at the end of `compute_logits`:

```python
logits = sharding_constraint_tensor(logits, self.mesh, (None, None))
```

PR #3820 gated that line off for 2D mesh (`if self.enable_tensor_parallel and not self.use_2d_mesh:`). Logits then enter the sampler graph still sharded along the `model` axis, vocab-dim reductions operate per-shard, sampler picks invalid tokens → garbage.

CPU sampling masks the bug because `logits.cpu()` forces an implicit all-gather.

## Fix

Drop the `not self.use_2d_mesh` guard so the constraint applies to both mesh shapes — restores pre-#3820 behavior for the producer side of the handoff.

## Why this is safe

- **Partial revert, not new behavior.** Pre-#3820 the constraint was unconditional. The 1D path has been running with it in production all along — known-safe.
- **The deleted comment said not to remove it.** PR #3820 removed this warning along with the 2D-path constraint:
  > Must be inside the compiled graph — external sharding_constraint between compiled functions breaks.
- **PR #3820 was asymmetric.** It added a `(None, None)` constraint at the end of `select_hidden_states` (2D-only) and *removed* the matching one at the end of `compute_logits` (2D-only). The fix only undoes the removal; the added one stays.
- **Blast radius:** one file, one line, no API change.

## Validation

Qwen3-8B before fix:

```
[2d_device] ' earnservice baise]=="earned cookie Res Html match revealslegt
             hostage practical revealsे� GitExpect록_ball Dumpster mijn hem
             Instpri()\\ PureComponent'
[1d_device] " park. I can't wait for the weekend. The green trees and the blue
             sky can make me feel relaxed. But the park"
```

Qwen3-8B after fix (2d_device now matches 1d_device word-for-word):

```
[2d_device] " park. I can't wait for the weekend. The green trees and the blue
             sky can make me feel relaxed. But the park"
[1d_device] " park. I can't wait for the weekend. The green trees and the blue
             sky can make me feel relaxed. But the park"
```

Qwen3-32B (the original failing case) now passes:

```
prompt: I like taking walks in the
output:  country, but sometimes I feel lonely. What can I do to make it
         more enjoyable? Also, I noticed some people in the
```
